lib/services/catalog/queries/index.js

```diff
@@ -192,6 +192,12 @@ module.exports = function (service) {
 					newObject.id = object.id;
 					newObject.spec = specId;
 
+					// Retain locally set relationships except for entities, those should be from the providers
+					if (_.has(object, 'relationships.entities')) {
+						delete object.relationships.entities;
+					}
+					newObject.relationships = _.merge({}, newObject.relationships, object.relationships);
```

Comment:
Because of the unexpected results with _.merge() and Arrays, maybe we just want to do a shallow copy here?

`newObject.relationships = Object.assign({}, newObject.relationships, object.relationships);`